includes/compat/trait-wc-stripe-subscriptions.php

```diff
@@ -1016,69 +1016,69 @@ public function maybe_render_subscription_payment_method( $payment_method_to_dis
 		$payment_method_to_display = __( 'N/A', 'woocommerce-gateway-stripe' );
 
 		try {
-			// Retrieve all possible payment methods for subscriptions.
-			foreach ( WC_Stripe_Customer::STRIPE_PAYMENT_METHODS as $payment_method_type ) {
-				foreach ( $stripe_customer->get_payment_methods( $payment_method_type ) as $source ) {
-					if ( $source->id !== $stripe_source_id ) {
-						continue;
-					}
+			// Retrieve all possible payment methods for the customer so we make minimal API calls to Stripe.
+			$customer_payment_methods = $stripe_customer->get_all_payment_methods();
+
+			foreach ( $customer_payment_methods as $source ) {
+				if ( $source->id !== $stripe_source_id ) {
+					continue;
+				}
+
+				// Legacy handling for Stripe Card objects. ref: https://docs.stripe.com/api/cards/object
+				if ( isset( $source->object ) && WC_Stripe_Payment_Methods::CARD === $source->object ) {
+					/* translators: 1) card brand 2) last 4 digits */
+					$payment_method_to_display = sprintf( __( 'Via %1$s card ending in %2$s', 'woocommerce-gateway-stripe' ), ( isset( $source->brand ) ? wc_get_credit_card_type_label( $source->brand ) : __( 'N/A', 'woocommerce-gateway-stripe' ) ), $source->last4 );
+					break;
+				}
 
-					// Legacy handling for Stripe Card objects. ref: https://docs.stripe.com/api/cards/object
-					if ( isset( $source->object ) && WC_Stripe_Payment_Methods::CARD === $source->object ) {
+				switch ( $source->type ) {
```

Comment:
I think we have the opportunity to move this to a new method. `maybe_render_subscription_payment_method` is exceptionally long.